Closes #814: Adding pages, events and some paragraph types, files, and taxonomy to demo content.

trackleft opened the pull request:
Adding content from https://live-azs-demo.pantheonsite.io/
## Description
Moved az_migrations plugin for paragraphs behaviors to az_paragraphs so we can share with the demo module.
Adding flexible page and most fields
Added pages to menu
Added text paragraphs

## Related Issue
#123
#121 

## Tools
Used this tool https://www.freeformatter.com/json-escape.html to escape html for use in the json files.

Test site: https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-879.probo.build/
Probo: https://app.probo.ci/#/dashboard/build/3faa452f-02dd-4fe8-8ebd-75f6e71a6c6e

danahertzberg commented:
Is this PR associated with this future ticket? https://github.com/az-digital/az_quickstart/issues/814

trackleft commented on modules/custom/az_demo/az_demo.install:
```diff
@@ -7,19 +7,38 @@
     $executable->import();
   }
+
+  // Set front page.
```
This might be controversal. Setting the front page via install hook.

trackleft commented on modules/custom/az_demo/migrations/az_demo_event_rrule.yml:
```diff
@@ -0,0 +1,96 @@
+dependencies:
+  enforced:
+    module:
+      - az_event
```
```suggestion
      - az_event
      
```

trackleft commented:
It might be good to put a warning on the demo content PR that says it will change their site, so don't do it on production.

joeparsons commented on modules/custom/az_demo/az_demo.install:
```diff
@@ -7,19 +7,33 @@
 function az_demo_install() {
   $tag = 'Quickstart Demo Content';
 
+  // Set default timezone for demo content.
+  $config = \Drupal::service('config.factory')->getEditable('system.date');
+  $config->set('timezone.default', 'America/Phoenix')->save();
+
```
I think this should no longer be necessary since we merged #891.

trackleft commented:
Fixed the issue with the ParagraphsBehavior.php file https://github.com/az-digital/az_quickstart/pull/879/commits/3d5f78ec702f59e4ce375830d8879b7a9449f3fc#diff-2ad805432b80cd56e2b81e226cdd0f570fa291eb1629be42f56e48a429534be0

Added a Warning to the module description: https://github.com/az-digital/az_quickstart/pull/879/commits/3d5f78ec702f59e4ce375830d8879b7a9449f3fc#diff-4bf94b4f22f721fd408d2419021ffa97dc33d0a170fd1ac6948f5dd226926d99R3

Added a PHPUnit test https://github.com/az-digital/az_quickstart/pull/879/commits/3d5f78ec702f59e4ce375830d8879b7a9449f3fc#diff-0e2f3271a3d28dd7fea03b4040bc7a03a634491648ea017149145c03f885e140R64-R75

camikazegreen commented on modules/custom/az_demo/tests/src/Functional/AZDemoContentTest.php:
```diff
@@ -53,15 +53,26 @@ protected function setUp() {
   }
 
+  /**
+   * Tests that imported utility links exist.
```
```suggestion
   * Tests that imported title is correct.
```

trackleft replied:
I forgot to commit my correct file.... This should be fixed.

trackleft commented:
FYI fixing the demo content migration broke the full-width bg migration in az_migrate
``` [warning] Illegal offset type ParagraphsBehavior.php:83
337
338 1/5 [=====>----------------------]  20% [warning] Illegal offset type ParagraphsBehavior.php:83
339
340 2/5 [===========>----------------]  40% [warning] Illegal offset type ParagraphsBehavior.php:83
341
342 3/5 [================>-----------]  60% [warning] Illegal offset type ParagraphsBehavior.php:83
343
344 4/5 [======================>-----]  80% [warning] Illegal offset type ParagraphsBehavior.php:83
345
346 5/5 [============================] 100% [notice] Processed 5 items (5 created, 0 updated, 0 failed, 0 ignored) - done with 'az_paragraph_content_marquee'
```
And this too 
``` [warning] Illegal offset type ParagraphsBehavior.php:83
350
351 1/5 [=====>----------------------]  20% [warning] Illegal offset type ParagraphsBehavior.php:83
352
353 2/5 [===========>----------------]  40% [warning] Illegal offset type ParagraphsBehavior.php:83
354
355 3/5 [================>-----------]  60% [warning] Illegal offset type ParagraphsBehavior.php:83
356
357 4/5 [======================>-----]  80% [warning] Illegal offset type ParagraphsBehavior.php:83
358
359 5/5 [============================] 100% [notice] Processed 5 items (5 created, 0 updated, 0 failed, 0 ignored) - done with 'az_paragraph_full_width_media_row'
```

Should be simple enough, just need to get a string from the source db instead of an array.

smayden reviewed (approved):
Tested front end. This looks good to me, at least for now since demo content is still experimental. 

Some future improvements for later:
- Need to add news from the demo site
- If we keep the annual events page, we'll need some events to be tagged as such (they were tagged on demo site where the page came from, but then events brought from QS1 I think)
- I'd change the homepage to not have the kitten info, but rather somehow indicate that demo content is turned on, what it includes, how to disable, etc.

trackleft commented:
> FYI fixing the demo content migration broke the full-width bg migration in az_migrate
> 
> ```
> 337
> 338 1/5 [=====>----------------------]  20% [warning] Illegal offset type ParagraphsBehavior.php:83
> 339
> 340 2/5 [===========>----------------]  40% [warning] Illegal offset type ParagraphsBehavior.php:83
> 341
> 342 3/5 [================>-----------]  60% [warning] Illegal offset type ParagraphsBehavior.php:83
> 343
> 344 4/5 [======================>-----]  80% [warning] Illegal offset type ParagraphsBehavior.php:83
> 345
> 346 5/5 [============================] 100% [notice] Processed 5 items (5 created, 0 updated, 0 failed, 0 ignored) - done with 'az_paragraph_content_marquee'
> ```
> 
> And this too
> 
> ```
> 350
> 351 1/5 [=====>----------------------]  20% [warning] Illegal offset type ParagraphsBehavior.php:83
> 352
> 353 2/5 [===========>----------------]  40% [warning] Illegal offset type ParagraphsBehavior.php:83
> 354
> 355 3/5 [================>-----------]  60% [warning] Illegal offset type ParagraphsBehavior.php:83
> 356
> 357 4/5 [======================>-----]  80% [warning] Illegal offset type ParagraphsBehavior.php:83
> 358
> 359 5/5 [============================] 100% [notice] Processed 5 items (5 created, 0 updated, 0 failed, 0 ignored) - done with 'az_paragraph_full_width_media_row'
> ```
> 
> Should be simple enough, just need to get a string from the source db instead of an array.

☝️ This is resolved.

joeparsons commented on modules/custom/az_paragraphs/src/Plugin/migrate/process/ParagraphsBehavior.php:
```diff
@@ -59,21 +63,24 @@ public function transform($value, MigrateExecutableInterface $migrate_executable
-        $behavior['bg_attachment'] = $bg_attach_mapping[$bg_attach_item['value']];
+      if (!empty($row->getSourceProperty($this->configuration['bg_attach']))) {
+        $bg_attach_mapping = [
+          'bg-attachment-fixed' => 'bg-fixed',
+          'bg-fixed' => 'bg-fixed',
```
I'm wondering why these 2 different `bg-fixed` source values are needed here.  Do the values in the demo content JSON files not match the D7 Quickstart value?

joeparsons replied:
> I'm wondering why these 2 different `bg-fixed` source values are needed here. Do the values in the demo content JSON files not match the D7 Quickstart value?

Never mind.  I think I see what's going on now.

joeparsons reviewed (approved):
I think we can merge this now.

joeparsons commented:
@tadean Do you have time to review this one?

trackleft commented on modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_content_marquee.yml:
```diff
@@ -20,34 +20,40 @@ process:
     plugin: default_value
     default_value: az_standard
 
+  field_uaqs_setting_bg_attach_value: field_uaqs_setting_bg_attach/0/value
```
`field_uaqs_setting_bg_attach/0/value`  could probably just be on line 54 instead of making this pseudo destination.
